examples/gno.land/p/demo/avl/rolist/rolist.gno

```diff
@@ -0,0 +1,119 @@
+// Package rolist provides a read-only wrapper for list.List with safe value transformation.
+//
+// It is useful when you want to expose a read-only view of a list while ensuring that
+// the sensitive data cannot be modified.
+//
+// Example:
+//
+//	// Define a user structure with sensitive data
+//	type User struct {
+//	    Name     string
+//	    Balance  int
+//	    Internal string // sensitive field
+//	}
+//
+//	// Create and populate the original list
+//	privateList := list.New()
+//	privateList.Append(&User{
+//	    Name:     "Alice",
+//	    Balance:  100,
+//	    Internal: "sensitive",
+//	})
```

Comment:
I'm wondering if we're deviating from what we generally want with `avl`. If someone wants to use a tree, let them use `avl` and its helpers. If not, they're welcome to use the native slice. I think people won't understand when to use the native slice vs this kind of package, with an underlying `avl` tree.

If the point of using this kind of package because of the "safe" functionality, maybe this package should be named differently & not be under `avl`.

Reply:
Yes and no. For instance, I've opened an alternative version of an efficient structure to manage lists here: #3407.

However, the comments on my PR suggest there may be valid reasons to use the `avl/list`. Additionally, the alternative aims to embrace the Unix philosophy and Golang composability, demonstrating that we can build layers upon layers.

I plan to use my #3407 alternative in most cases. However, some people are still using a standard AVL map as a list; I believe you're doing this too for the `hof` realm, for example. Whenever you see someone using an avl.Tree with a seqid, there's a good chance they will prefer this package. For others, my `ulist` alternative might be a better fit.

I believe we should have such packages, but we need to:
1. Include great comments (let me know if I should improve them here or elsewhere).
2. Provide excellent examples—let's create more `r/docs`, including one comprehensive guide for all the data structures for easier comparison.